nestkernel/event_delivery_manager.cpp

```diff
@@ -384,12 +392,21 @@ EventDeliveryManager::gather_spike_data_( const thread tid,
 #pragma omp single
       {
         buffer_size_spike_data_has_changed_ = kernel().mpi_manager.increase_buffer_size_spike_data();
+        decrease_buffer_size_spike_data_ = false;
       }
     }
 #pragma omp barrier
 
   } // of while
 
+#pragma omp single
```

Comment:
if i understand the algorithm correctly, then the buffers shrink after every communication round if no more than one round was required. doesn't this mean the buffers always shrink below the required size, then are increased again only to continue shrinking? given the choices for increase and decrease factors (for fixed number of spikes) this seems to imply a cycle length of ~5. have you thought about possibilities to reduce/remove this periodic increasing and decreasing of buffer sizes?

Reply:
Yes, buffer size will oscillate around some value (assuming network activity is steady). Note, the 'shrink factor' was derived sometime ago, and may not be optimum (although I suspect this is case dependent). The desire is to vary the count argument of Alltoall by tracking network activity; we want to minimise the communication of 'padding' in the buffer (while avoiding use of Alltoallv). There have been many (very different) versions of the logic presented here, but these come with additional overhead associated with synchronisation, or some amount of compute. This version is embarrassingly simple, efficient and robust. Since only the count variable is modified at the end of a communication round (assuming that our communication vectors already have a size that is greater than necessary, otherwise `vector::resize` is invoked), this oscillation does not incur a high cost (we may have some additional communication rounds).